external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -148,17 +149,31 @@ private[kinesis] class KinesisReceiver[T](
 
     kinesisCheckpointer = new KinesisCheckpointer(receiver, checkpointInterval, workerId)
     val kinesisProvider = kinesisCreds.provider
-    val kinesisClientLibConfiguration = new KinesisClientLibConfiguration(
-          checkpointAppName,
-          streamName,
-          kinesisProvider,
-          dynamoDBCreds.map(_.provider).getOrElse(kinesisProvider),
-          cloudWatchCreds.map(_.provider).getOrElse(kinesisProvider),
-          workerId)
-        .withKinesisEndpoint(endpointUrl)
-        .withInitialPositionInStream(initialPositionInStream)
-        .withTaskBackoffTimeMillis(500)
-        .withRegionName(regionName)
+
+    val kinesisClientLibConfiguration = {
+      var clientLibConf = new KinesisClientLibConfiguration(
+        checkpointAppName,
+        streamName,
+        kinesisProvider,
+        dynamoDBCreds.map(_.provider).getOrElse(kinesisProvider),
+        cloudWatchCreds.map(_.provider).getOrElse(kinesisProvider),
+        workerId)
+      .withKinesisEndpoint(endpointUrl)
+      .withTaskBackoffTimeMillis(500)
+      .withRegionName(regionName)
+
+      /** Enhance the Kinesis receiver based on InitialPositionInStream  */
```

Comment:
I'd use the standard ```// comment``` syntax here instead of JavaDoc as you aren't documenting a class or method.